src/Tribe/Attendees.php

```diff
@@ -565,18 +608,36 @@ public function maybe_generate_csv() {
 			return;
 		}
 
-		if ( ! wp_verify_nonce( $_GET['attendees_csv_nonce'], 'attendees_csv_nonce' ) || ! $this->user_can( 'edit_posts', $_GET['event_id'] ) ) {
+		$event_id = absint( $_GET['event_id'] );
+
+		// Verify event ID is a valid integer and the nonce is accepted.
+		if ( empty( $event_id ) || ! wp_verify_nonce( $_GET['attendees_csv_nonce'], 'attendees_csv_nonce' ) ) {
 			return;
 		}
 
+		$event = get_post( $event_id );
+
+		// Verify event exists and current user has access to it.
+		if (
+			! $event instanceof WP_Post
+			|| ! $this->user_can( 'edit_posts', $event_id )
+		) {
+			return;
+		}
+
+		// Generate filtereed list of attendees.
+		$items = $this->generate_filtered_list( $event_id );
+
+		// Sanitize items for CSV usage.
+		$items = array_map( [ $this, 'sanitize_csv_rows' ], $items );
```

Comment:
I wonder if we should sanitize after the filter?

Reply:
+1

Reply:
I left this before the filter to allow people to allow customized formulas for their attendees exports. I can put it after the filter if you think we shouldn't allow that at all.